src/plugin/__tests__/element.spec.ts

```diff
@@ -350,6 +357,45 @@ describe("buildElementPlugin", () => {
       },
     });
 
+    type ExternalData = { nestedElementValues: { field1: string } };
+    type OtherExternalData = { otherValues: { field1: string } };
+
+    const testElementWithTransform = createElementSpec(
+      {
+        field1: { type: "richText" },
+      },
+      () => null,
+      () => null,
+      {
+        transformElementDataIn: (external: ExternalData) => {
+          return external.nestedElementValues;
+        },
+        transformElementDataOut: (internal): ExternalData => {
+          return {
+            nestedElementValues: internal,
+          };
```

Comment:
Im assuming this will reshape/redefine the JSON response for CAPI?
- Deal with data output in flexible content: 
```json
  {
    elementType: "pullquote",
    fields: {
      html: "Example pullquote",
      role: "supporting",
      attribution: "Attribution"
    },
    assets: [ ]
  },
```

This is a great idea to centralise and standardise it.

Reply:
Exactly, this would be a means to transform the data structure used by ProseMirror to one expected by CAPI.